hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkFileUtility.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.hadoop.ozone.debug;
+
+import org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos;
+import org.apache.hadoop.hdds.scm.container.common.helpers.StorageContainerException;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.apache.hadoop.ozone.container.common.impl.ChunkLayOutVersion;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+
+import java.io.File;
+
+
+public final class ChunkFileUtility {
+
+    static String getChunkLocationPath(String containerLocation) {
+        return containerLocation + File.separator + OzoneConsts.STORAGE_DIR_CHUNKS;
+    }
+
+    public static String getChunkFilePath(ContainerProtos.ChunkInfo
+      chunkInfo, OmKeyLocationInfo keyLocation,
+      ContainerProtos.ContainerDataProto data,
+      ChunkLayOutVersion layOutVersion)
+      throws StorageContainerException {
+        switch (layOutVersion) {
+        case FILE_PER_CHUNK:
+          return  getChunkLocationPath(data
+                  .getContainerPath())
+                  + File.separator
+                  + chunkInfo.getChunkName();
+        case FILE_PER_BLOCK:
+          return  getChunkLocationPath(data
+                  .getContainerPath())
+                  + File.separator
+                  + keyLocation.getLocalID() + ".block";
+        default:
+          throw new StorageContainerException("chunk strategy does not exist",
+                  ContainerProtos.Result.UNABLE_TO_FIND_CHUNK);
+        }
```

Comment:
I'd prefer taking chunk file path directly from `ChunkLayOutVersion.getChunkFile()` instead of the `switch`.  This way if a new enum constant is introduced, this code will work without any changes.

This needs a minor refactoring in `ChunkLayOutVersion`, which accepts a `ContainerData` object, to work directly with the chunk dir:

```diff
--- hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ChunkLayOutVersion.java
+++ hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ChunkLayOutVersion.java
@@ -40,17 +40,15 @@

   FILE_PER_CHUNK(1, "One file per chunk") {
     @Override
-    public File getChunkFile(ContainerData containerData, BlockID blockID,
-        ChunkInfo info) throws StorageContainerException {
-      File chunksLoc = verifyChunkDirExists(containerData);
-      return chunksLoc.toPath().resolve(info.getChunkName()).toFile();
+    public File getChunkFile(File chunkDir, BlockID blockID,
+        ChunkInfo info) {
+      return chunkDir.toPath().resolve(info.getChunkName()).toFile();
     }
   },
   FILE_PER_BLOCK(2, "One file per block") {
     @Override
-    public File getChunkFile(ContainerData containerData, BlockID blockID,
-        ChunkInfo info) throws StorageContainerException {
-      File chunkDir = verifyChunkDirExists(containerData);
+    public File getChunkFile(File chunkDir, BlockID blockID,
+        ChunkInfo info) {
       return new File(chunkDir, blockID.getLocalID() + ".block");
     }
   };
@@ -118,8 +116,14 @@ public String getDescription() {
     return description;
   }

-  public abstract File getChunkFile(ContainerData containerData,
-      BlockID blockID, ChunkInfo info) throws StorageContainerException;
+  public abstract File getChunkFile(File chunksDir,
+      BlockID blockID, ChunkInfo info);
+
+  public File getChunkFile(ContainerData containerData,
+      BlockID blockID, ChunkInfo info) throws StorageContainerException {
+    File chunksLoc = verifyChunkDirExists(containerData);
+    return getChunkFile(chunksLoc, blockID, info);
+  }

   @Override
   public String toString() {
```

Reply:
Thankyou @adoroszlai for the review. Addressed review comments .